include/sound/sof/ipc4/header.h

```diff
@@ -0,0 +1,324 @@
+#define SOF_IPC4_MOD_EXT_DOMAIN(x)		((x) << SOF_IPC4_MOD_EXT_DOMAIN_SHIFT)
+
+/*  bind/unbind module ipc msg */
+#define SOF_IPC4_MOD_EXT_DST_MOD_ID_SHIFT	0
+#define SOF_IPC4_MOD_EXT_DST_MOD_ID_MASK	GENMASK(15, 0)
+#define SOF_IPC4_MOD_EXT_DST_MOD_ID(x)		((x) << SOF_IPC4_MOD_EXT_DST_MOD_ID_SHIFT)
+
+#define SOF_IPC4_MOD_EXT_DST_MOD_INSTANCE_SHIFT	16
+#define SOF_IPC4_MOD_EXT_DST_MOD_INSTANCE_MASK	GENMASK(23, 16)
+#define SOF_IPC4_MOD_EXT_DST_MOD_INSTANCE(x)	((x) << SOF_IPC4_MOD_EXT_DST_MOD_INSTANCE_SHIFT)
+
+#define SOF_IPC4_MOD_EXT_DST_MOD_QUEUE_ID_SHIFT	24
+#define SOF_IPC4_MOD_EXT_DST_MOD_QUEUE_ID_MASK	GENMASK(26, 24)
+#define SOF_IPC4_MOD_EXT_DST_MOD_QUEUE_ID(x)	((x) << SOF_IPC4_MOD_EXT_DST_MOD_QUEUE_ID_SHIFT)
+
+#define SOF_IPC4_MOD_EXT_SRC_MOD_QUEUE_ID_SHIFT	27
+#define SOF_IPC4_MOD_EXT_SRC_MOD_QUEUE_ID_MASK	GENMASK(29, 27)
+#define SOF_IPC4_MOD_EXT_SRC_MOD_QUEUE_ID(x)	((x) << SOF_IPC4_MOD_EXT_SRC_MOD_QUEUE_ID_SHIFT)
+
+#define MOD_ENABLE_LOG	6
+#define MOD_SYSTEM_TIME	20
+
+/* set module large config */
+#define SOF_IPC4_MOD_EXT_MSG_SIZE_SHIFT		0
+#define SOF_IPC4_MOD_EXT_MSG_SIZE_MASK		GENMASK(19, 0)
+#define SOF_IPC4_MOD_EXT_MSG_SIZE(x)		((x) << SOF_IPC4_MOD_EXT_MSG_SIZE_SHIFT)
+
+#define SOF_IPC4_MOD_EXT_MSG_PARAM_ID_SHIFT	20
+#define SOF_IPC4_MOD_EXT_MSG_PARAM_ID_MASK	GENMASK(27, 20)
+#define SOF_IPC4_MOD_EXT_MSG_PARAM_ID(x)	((x) << SOF_IPC4_MOD_EXT_MSG_PARAM_ID_SHIFT)
+
+#define SOF_IPC4_MOD_EXT_MSG_LAST_BLOCK_SHIFT	28
+#define SOF_IPC4_MOD_EXT_MSG_LAST_BLOCK_MASK	BIT(28)
+#define SOF_IPC4_MOD_EXT_MSG_LAST_BLOCK(x)	((x) << SOF_IPC4_MOD_EXT_MSG_LAST_BLOCK_SHIFT)
+
+#define SOF_IPC4_MOD_EXT_MSG_FIRST_BLOCK_SHIFT	29
+#define SOF_IPC4_MOD_EXT_MSG_FIRST_BLOCK_MASK	BIT(29)
+#define SOF_IPC4_MOD_EXT_MSG_FIRST_BLOCK(x)	((x) << SOF_IPC4_MOD_EXT_MSG_FIRST_BLOCK_SHIFT)
+
+/* ipc4 notification msg */
+#define SOF_IPC4_GLB_NOTIFY_TYPE_SHIFT		16
+#define SOF_IPC4_GLB_NOTIFY_TYPE_MASK		0xFF
+#define SOF_IPC4_GLB_NOTIFY_TYPE(x)		(((x) >> SOF_IPC4_GLB_NOTIFY_TYPE_SHIFT) \
+						& SOF_IPC4_GLB_NOTIFY_TYPE_MASK)
+
+#define SOF_IPC4_GLB_NOTIFY_MSG_TYPE_SHIFT	24
+#define SOF_IPC4_GLB_NOTIFY_MSG_TYPE_MASK	0x1F
+#define SOF_IPC4_GLB_NOTIFY_MSG_TYPE(x)		(((x) >> SOF_IPC4_GLB_NOTIFY_MSG_TYPE_SHIFT) \
+						& SOF_IPC4_GLB_NOTIFY_MSG_TYPE_MASK)
+
+enum cavs_ipc_notification_type {
+	/* Phrase detected (notification from WoV module) */
+	SOF_IPC4_GLB_NOTIFY_PHRASE_DETECTED = 4,
+	/*Event from a resource (pipeline or module instance) */
+	SOF_IPC4_GLB_NOTIFY_RESOURCE_EVENT = 5,
+	/* Debug log buffer status changed */
+	SOF_IPC4_GLB_NOTIFY_LOG_BUFFER_STATUS = 6,
+	/* Timestamp captured at the link */
+	SOF_IPC4_GLB_NOTIFY_TIMESTAMP_CAPTURED = 7,
+	/* FW complete initialization */
+	SOF_IPC4_GLB_NOTIFY_FW_READY = 8,
+	/* Audio classifier result (ACA) */
+	SOF_IPC4_GLB_NOTIFY_FW_AUD_CLASS_RESULT = 9,
+	/* Exception caught by DSP FW */
+	SOF_IPC4_GLB_NOTIFY_EXCEPTION_CAUGHT = 10,
+	/* Custom module notification */
+	SOF_IPC4_GLB_NOTIFY_MODULE_NOTIFICATION = 12,
```

Comment:
add a comment on 11 being skipped

Reply:
> add a comment on 11 being skipped

That was copied from the existing cavs firmware. I have no idea why it starts with 4 and skip 11. Is `"To be compatible with the existing cavs firmware"` ok?